src/main/java/de/bonndan/nivio/util/URLHelper.java

```diff
@@ -23,40 +25,40 @@ public static boolean isLocal(URL url) {
      * Tries to create an URL if a path is given.
      *
      * @param url string url or local path to file
-     * @return an URL or null
+     * @return an Optional of URL (empty on malformed urls)
      */
-    @Nullable
-    public static URL getURL(String url) {
+    public static Optional<URL> getURL(String url) {
         if (url == null) {
-            return null;
+            return Optional.empty();
         }
         try {
-            return new URL(url);
+            return Optional.of(new URL(url));
         } catch (MalformedURLException e) {
             File file = new File(url);
             if (file.exists() && !url.startsWith("file:")) {
                 try {
-                    return file.toURI().toURL();
+                    return Optional.of(file.toURI().toURL());
                 } catch (MalformedURLException ignored) {
-                    return null;
                 }
             }
         }
-        return null;
+        return Optional.empty();
     }
 
     /**
      * Returns the path of a file as URL or null
      */
-    public static URL getParentPath(String url) {
-        URL url1 = getURL(url);
-        if (url1 == null) {
-            return null;
+    public static Optional<URL> getParentPath(String url) {
+
+        Optional<URL> url1 = getURL(url);
+        if (url1.isEmpty()) {
+            return Optional.empty();
         }
+
         try {
-            return url1.toURI().resolve(".").toURL();
+            return Optional.of(url1.get().toURI().resolve(".").toURL());
```

Comment:
Could also be:

    url1.map(url -> url.toURI().resolve(".").toURL());

That would make the previous empty check unnecessary.

Reply:
👍